evm/utils/state_access_restriction.py

```diff
@@ -23,6 +23,32 @@ def is_accessible(address, slot_as_key, access_prefix_list):
     return False
 
 
+def remove_redundant_prefixes(prefix_list):
+    """Given a list of strings, this removes redundant strings that are covered
+    by other strings in the list. For example, given `["eth", "ethereum"]`,
+    this function will return just `["ethereum"]` as every string with
```

Comment:
This is the wrong way around: `"eth"` should stay (more general), `"ethereum"` should be removed (more specific).

Reply:
Derp, dunno how I misunderstood the whole thing. Anyway, I've fixed this and added some tests as a sanity check.